eng/pipelines/templates/stages/archetype-net-release.yml

```diff
@@ -0,0 +1,246 @@
+parameters:
+  Artifacts: []
+  ArtifactName: 'not-specified'
+
+stages:
+  - stage: Signing
+    dependsOn: ${{parameters.DependsOn}}
+    jobs:
+      - deployment: SignPackage
+        environment: esrp
+        timeoutInMinutes: 20
+        pool:
+          vmImage: vs2017-win2016
+
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - checkout: none
+
+                - download: current
+                  artifact: ${{parameters.ArtifactName}}
+                  timeoutInMinutes: 5
+
+                - ${{ each artifact in parameters.Artifacts }}:
+                  - pwsh: |
+                      New-Item -Type Directory -Name staging -Path $(Pipeline.Workspace) -Force
+                      Copy-Item $(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.name}}.[0-9]*.[0-9]*.[0-9]* $(Pipeline.Workspace)/staging
+                      Get-ChildItem $(Pipeline.Workspace)/staging
+                    displayName: Copying ${{artifact.name}} to staging directory
+
+                - template: pipelines/steps/sign-package.yml@azure-sdk-build-tools
+                  parameters:
+                    ArtifactName: ${{parameters.ArtifactName}}
```

Comment:
I assume this should now be a directory and not a artifactname.

Reply:
this is just the value of all the packages which is just `packages`. The artifacts are all copied to `$(Pipeline.Workspace)/staging` which is used in various steps in the template. Should I pass that as its own parameter.
i.e. `StagingDirectory: $(Pipeline.Workspace)/staging`

Reply:
Oh I see it is used for the name of the output artifact. It is probably worth moving that out of the signing template as well and if a pipeline wants to they can publish the signed artifacts themselves.

For the signing template I think all we should need is the directory of files to sign, which is what net-signing.yml already does. By the way can you please try to reconcile your new template and https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools?path=%2Fpipelines%2Fsteps%2Fnet-signing.yml&version=GBMakeEngSysToolsPublic? We should need only one.